### PR TITLE
78 fine tune color recognition

### PR DESCRIPTION
**Pull request by Greve2001**
I have changed the resolution and parameters to suit it.

### CODE REVIEW

File: remote/src/main/java/vision/Detection.java [hunk 1]
```diff
@@ -224,13 +245,14 @@ private Optional<Ball> findOrangeBall(Mat frame) {
 
         // Create a mask to seperate the orange ball
         Mat mask = new Mat();
-        Scalar lower = new Scalar(11, 50, 220);
-        Scalar upper = new Scalar(30, 240, 255);
-        Core.inRange(frameHsv, lower, upper, mask);
+        Core.inRange(frameHsv, lOrangeBall, uOrangeBall, mask);
+
+        // Show Mask
+        //HighGui.imshow("orangeMask", mask);
```
Comment (yech199) [inline remote/src/main/java/vision/Detection.java:251]: Why do we need to show this?

File: remote/src/main/java/vision/Detection.java [hunk 2]
```diff
@@ -224,13 +245,14 @@ private Optional<Ball> findOrangeBall(Mat frame) {
...
 
         // Apply blur for noise reduction
         Mat frameBlur = new Mat();
-        Imgproc.GaussianBlur(mask, frameBlur, new Size(9,9), 0);
+        Imgproc.GaussianBlur(mask, frameBlur, new Size(7,7), 0);
```
Comment (yech199) [inline remote/src/main/java/vision/Detection.java:255]: Comment/Document why this value have been chosen. It kind of feels like magic values...

File: remote/src/main/java/vision/Detection.java [hunk 3]
```diff
@@ -260,21 +282,16 @@ private Optional<Ball> findOrangeBall(Mat frame) {
      * @return Returns array of points for the 2 markers.
      */
     public Point[] getRotationCoordsFromFrame(Mat frame) {
-        final int areaLowerThreshold = 60;
-        final int areaUpperThreshold = 350;
+        final int areaLowerThreshold = 100;
+        final int areaUpperThreshold = 1000;
```
Comment (yech199) [inline remote/src/main/java/vision/Detection.java:286]: Comment/Document why these values have been chosen. It kind of feels like magic values...

File: remote/src/main/java/vision/Detection.java [hunk 4]
```diff
@@ -260,21 +282,16 @@ private Optional<Ball> findOrangeBall(Mat frame) {
...
 
         // Transform frame
-        Mat frameHSV = new Mat();
         Mat frameBlur = new Mat();
-
-        Imgproc.cvtColor(frame, frameHSV, Imgproc.COLOR_BGR2HSV);
-        Imgproc.GaussianBlur(frameHSV, frameBlur, new Size(7,7), 7, 0);
+        Imgproc.GaussianBlur(frame, frameBlur, new Size(7,7), 7, 0);
```
Comment (yech199) [inline remote/src/main/java/vision/Detection.java:290]: Comment/Document why these values have been chosen. It kind of feels like magic values...

File: remote/src/main/java/vision/Detection.java [hunk 5]
```diff
@@ -260,21 +282,16 @@ private Optional<Ball> findOrangeBall(Mat frame) {
...
 
         // Create a mask
         Mat mask = new Mat();
-        Scalar lower = new Scalar(100, 100, 120);
-        Scalar upper = new Scalar(255, 255, 255);
-        Core.inRange(frameBlur, lower, upper, mask);
+        Core.inRange(frameBlur, lRobot, uRobot, mask);
```
Comment (yech199) [inline remote/src/main/java/vision/Detection.java:294]: Comment/Document why these values have been chosen. It kind of feels like magic values...

**Review by yech199 — CHANGES_REQUESTED** [review]
We need more comments. Please comment magic values :)

File: remote/src/main/java/Main.java [hunk 6]
```diff
@@ -13,7 +13,7 @@ public static void main(String[] args) throws InterruptedException, MissingArgum
         }
 
         RobotController controller = new RobotController(args[0]); // Args[0] being and IP address
-        Detection detection = new Detection(0);
+        Detection detection = new Detection(2);
```
[the remark this answers is not in the document]
Comment (Greve2001) [reply remote/src/main/java/Main.java:16]: Get it!

**Review by Greve2001 — COMMENTED** [review]
PLEASE APPROVE :pray:

File: remote/src/main/java/vision/Detection.java [hunk 7]
```diff
@@ -108,8 +126,9 @@ private void detectCourse(VideoCapture capture) {
 
             debugGUI(debugFrame);
 
+            // Show debug frame
             HighGui.imshow("frame", debugFrame);
-            HighGui.waitKey(1);
+            HighGui.waitKey(1); // Wait 1ms per frame
```
Comment (yech199) [inline remote/src/main/java/vision/Detection.java:131]: Why 1ms?

File: remote/src/main/java/vision/Detection.java (hunk 4 above)
Comment (yech199) [reply remote/src/main/java/vision/Detection.java:294]: This is still magic

**Pull request merged**